third_party_release

```diff
@@ -1,10 +1,19 @@
+#
+# MAINTAINER
+#
+# When changing (add, upgrade, remove) dependencies
+# in this file for the CMake build,
+# please update the Bazel build as well,
+# see <root>/bazel/repository.bzl
+#
+
 gRPC=v1.43.2
 thrift=0.14.1
 abseil=20210324.0
 benchmark=v1.5.3
 googletest=release-1.10.0-459-ga6dfd3ac
 ms-gsl=v3.1.0-67-g6f45293
-nlohmann-json=v3.9.1
+nlohmann-json=v3.10.5
```

Comment:
I think for cmake, we should also update the nlohmann-json submodule in the third-party directory.

Reply:
Thanks @lalitb, and agree.

Talking about confusing, `cmake/nlohmann-json.cmake` uses the tags derived from `third_party_release` (`${nlohmann-json}`) and makes a git pull into ... the git submodule directory in `third_party`.

Updated the nlohmann_json submodule to v3.10.5.
Marking as resolved

Reply:
yes agree, I missed that :). So the submodule version would be used if the `cmake` build is done from the otel-cpp repo, otherwise `cmake/nlohmann-json.cmake` is used.